packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -78,26 +78,27 @@ internal fun <T, R> channelResultCallback(
 internal fun convertSyncError(syncError: SyncError): SyncException {
     val errorCode = syncError.errorCode
     val message = createMessageFromSyncError(errorCode)
-    return when (errorCode.errorCode) {
-        ErrorCode.RLM_ERR_WRONG_SYNC_TYPE -> WrongSyncTypeException(message)
+    return if (syncError.isFatal) {
+        // An unrecoverable exception happened
+        UnrecoverableSyncException(message)
+    } else {
+        when (errorCode.errorCode) {
+            ErrorCode.RLM_ERR_WRONG_SYNC_TYPE -> WrongSyncTypeException(message)
 
-        ErrorCode.RLM_ERR_INVALID_SUBSCRIPTION_QUERY -> {
-            // Flexible Sync Query was rejected by the server
-            BadFlexibleSyncQueryException(message)
-        }
-        ErrorCode.RLM_ERR_SYNC_COMPENSATING_WRITE -> CompensatingWriteException(message, syncError.compensatingWrites)
+            ErrorCode.RLM_ERR_INVALID_SUBSCRIPTION_QUERY -> {
+                // Flexible Sync Query was rejected by the server
+                BadFlexibleSyncQueryException(message)
+            }
 
-        ErrorCode.RLM_ERR_SYNC_PROTOCOL_INVARIANT_FAILED,
```

Comment:
Are these 3 all now under Fatal `UnrecoverableSyncException`?

Reply:
At least `RLM_ERR_SYNC_PERMISSION_DENIED` isn't, but shouldn't now any fatal (unrecoverable) errors have`is_fatal` set to true?

Reply:
Checked with core, `is_fatal` is a better property to identify unrecoverable errors.

Reply:
Seems safest to keep this, unless you are absolutely certain that core throws the three error codes with `is_fatal` set. If not we will suddenly throw them as `SyncExceptions` instead.